core/src/test/java/dev/morphia/test/mapping/TestReferenceAnnotationPosition.java

```diff
@@ -0,0 +1,136 @@
+package dev.morphia.test.mapping;
+
+import static java.lang.annotation.ElementType.*;
+import static java.lang.annotation.RetentionPolicy.*;
+import static org.junit.Assert.*;
+
+import java.lang.annotation.*;
+import java.util.*;
+
+import com.mongodb.client.*;
+import dev.morphia.annotations.*;
+import dev.morphia.test.*;
+import org.bson.*;
+import org.testng.annotations.*;
+
+@Target({FIELD})
+@Retention(RUNTIME)
+@interface AnAnnotation {
+
+}
+
+public class TestReferenceAnnotationPosition extends TestBase {
+
+	@BeforeMethod
+	public void setUp() {
+		getDs().getMapper().getCollection(Box.class).deleteMany(new Document());
+		getDs().getMapper().getCollection(Item.class).deleteMany(new Document());
+	}
+
+	@Test
+	public void testReferenceAnnotationShouldNeverEmbedDocuments() {
+		getDs().getMapper().map(Box.class, Item.class);
+
+		Item item1 = new Item("item one");
+		Item item2 = new Item("item two");
+		Box box = new Box();
+		box.setItems(Set.of(item1, item2));
+		box.setMoreItems(Set.of(item1, item2));
+		getDs().save(List.of(item1, item2));
+		getDs().save(box);
+
+		MongoCollection<Document> collection = getMapper()
+			.getCollection(Box.class)
+			.withDocumentClass(Document.class);
+
+		Document firstDoc = collection.find().first();
+		Object items = firstDoc.get("items");
+		Object moreItems = firstDoc.get("moreItems");
+
+		assertEquals("Items should be stored as the same documents", items, moreItems);
+	}
+
+}
+
+@Entity
+class Box {
+	@Id
+	private UUID id = UUID.randomUUID();
+	@Reference
+	@AnAnnotation
```

Comment:
What it is this?

Reply:
I've moved this annotation down, next to test entities (autoformatting again...) and added a small comment.

Basically, this @AnAnnotation does nothing beside separating the @Reference from the field so it's not the last annotation processed when looking for handlers. Overriding with `null`s in the loop makes the entities embedded, not referenced.

Reply:
Oh, i see.  you're trying to recreate the error condition.  that makes sense.

Reply:
Yes. I didn't want to use any 3rd party annotation, because this could make future readers that the type of the annotation matters.